src/arkode/arkode_splittingstep_coefficients.c

```diff
@@ -0,0 +1,474 @@
+    {
+      coefficients->beta[1][i + 1][j] = SUN_RCONST(1.0);
+    }
+  }
+
+  return coefficients;
+}
+
+/*---------------------------------------------------------------
+  Routine to construct a 3rd order method of Suzuki of the form
+  L(p1 h) * L*(p2 h) * L(p3 h) * L*(p4 h) * L(p5 h)
+  where L is a Lie-Trotter splitting and L* is its adjoint.
+  Composition is denoted by *.
+  ---------------------------------------------------------------*/
+SplittingStepCoefficients SplittingStepCoefficients_ThirdOrderSuzuki(
+  const int partitions)
+{
+  const SplittingStepCoefficients coefficients =
+    SplittingStepCoefficients_Alloc(1, 2 * partitions - 1, partitions);
+  if (coefficients == NULL) { return NULL; }
+
+  coefficients->order    = 3;
+  coefficients->alpha[0] = SUN_RCONST(1.0);
+
+  for (int i = 1; i < partitions; i++)
+  {
+    for (int j = 0; j < partitions; j++)
+    {
+      // Constants from https://doi.org/10.1143/JPSJ.61.3015 pg. 3019
+      const sunrealtype p1 =
+        SUN_RCONST(0.2683300957817599249569552299254991394812);
+      const sunrealtype p2 =
+        SUN_RCONST(0.6513314272356399320939424082278836500821);
+
+      coefficients->beta[0][i][j] = i + j < partitions ? p1 : (p1 + p2);
+      coefficients->beta[0][partitions + i - 1][j] =
+        SUN_RCONST(1.0) - (i + j < partitions ? (p1 + p2) : p1);
+    }
+  }
+
+  for (int i = 0; i < partitions; i++)
+  {
+    coefficients->beta[0][2 * partitions - 1][i] = SUN_RCONST(1.0);
+  }
+
+  return coefficients;
+}
+
+/*---------------------------------------------------------------
+  Routine to construct a composition method of the form
+  S(gamma_0 h)^c * S(gamma_1 h) * S(gamma_0)^c
+  where S is a lower order splitting (with Stang as the base case),
+  * and ^ denote composition, and c = composition_stages. This
+  covers both the triple jump (c=1) and Suzuki fractal (c=2).
+  ---------------------------------------------------------------*/
+static sunrealtype* const* SplittingStepCoefficients_ComposeStrangHelper(
+  const int partitions, const int order, const int composition_stages,
+  const sunrealtype start, const sunrealtype end, sunrealtype* const* const beta)
+{
+  const sunrealtype diff = end - start;
+  if (order == 2)
+  {
+    /* The base case is an order 2 Strang splitting */
+    const sunrealtype mid = start + diff / SUN_RCONST(2.0);
+    for (int j = 1; j <= partitions; j++)
+    {
+      for (int k = 0; k < partitions; k++)
+      {
+        beta[j][k] = (k + j < partitions) ? mid : end;
+      }
+    }
+
+    return &beta[partitions - 1];
+  }
+
+  sunrealtype* const* beta_cur = beta;
+  sunrealtype start_cur        = start;
+  /* This is essentially the gamma coefficient from Geometric Numerical
+   * Integration (https://doi.org/10.1007/3-540-30666-8) pg 44-45 scaled by the
+   * current interval */
+  const sunrealtype gamma =
+    diff / (composition_stages - 1 -
+            SUNRpowerR(composition_stages - 1, SUN_RCONST(1.0) / (order - 1)));
```

Comment:
Does this pass the CI without a "conversion error"?
```suggestion
            SUNRpowerR((sunrealtype) (composition_stages - 1), SUN_RCONST(1.0) / (order - 1)));
```

Reply:
Yes, this passes as is, so I don't thing the cast is necessary.

Reply:
In that case, you can feel free to ignore some of my other related comments as well.